OpenSim/Common/TimeSeriesTable.h

```diff
@@ -0,0 +1,158 @@
+/* -------------------------------------------------------------------------- *
+ *                            OpenSim:  TimeSeriesTable.h                     *
+ * -------------------------------------------------------------------------- *
+ * The OpenSim API is a toolkit for musculoskeletal modeling and simulation.  *
+ * See http://opensim.stanford.edu and the NOTICE file for more information.  *
+ * OpenSim is developed at Stanford University and supported by the US        *
+ * National Institutes of Health (U54 GM072970, R24 HD065690) and by DARPA    *
+ * through the Warrior Web program.                                           *
+ *                                                                            *
+ * Copyright (c) 2005-2015 Stanford University and the Authors                *
+ * Authors:                                                                   *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+#ifndef OPENSIM_TIME_SERIES_DATA_TABLE_H_
+#define OPENSIM_TIME_SERIES_DATA_TABLE_H_
+
+/** \file
+This file defines the TimeSeriesTable_ class, which is used by OpenSim to 
+provide an in-memory container for data access and manipulation.              */
+
+#include "OpenSim/Common/DataTable.h"
+
+
+namespace OpenSim {
+
+class InvalidTable : public Exception {
+public:
+    using Exception::Exception;
+};
+
+class TimeColumnNotIncreasing : public InvalidTable {
+public:
+    TimeColumnNotIncreasing(const std::string& file, 
+                            size_t line,
+                            const std::string& func) :
+        InvalidTable(file, line, func) {
+        std::string msg = "Time column is not strictly increasing";
+
+        addMessage(msg);
```

Comment:
Most users don't know anything about the C++ files, but we should still print it for when a user sends their log to someone who is more knowledgeable.